plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/MockThriftHttpMetastoreServer.java

```diff
@@ -0,0 +1,1763 @@
+import io.trino.hive.thrift.metastore.WMCreateOrDropTriggerToPoolMappingRequest;
+import io.trino.hive.thrift.metastore.WMCreateOrDropTriggerToPoolMappingResponse;
+import io.trino.hive.thrift.metastore.WMCreateOrUpdateMappingRequest;
+import io.trino.hive.thrift.metastore.WMCreateOrUpdateMappingResponse;
+import io.trino.hive.thrift.metastore.WMCreatePoolRequest;
+import io.trino.hive.thrift.metastore.WMCreatePoolResponse;
+import io.trino.hive.thrift.metastore.WMCreateResourcePlanRequest;
+import io.trino.hive.thrift.metastore.WMCreateResourcePlanResponse;
+import io.trino.hive.thrift.metastore.WMCreateTriggerRequest;
+import io.trino.hive.thrift.metastore.WMCreateTriggerResponse;
+import io.trino.hive.thrift.metastore.WMDropMappingRequest;
+import io.trino.hive.thrift.metastore.WMDropMappingResponse;
+import io.trino.hive.thrift.metastore.WMDropPoolRequest;
+import io.trino.hive.thrift.metastore.WMDropPoolResponse;
+import io.trino.hive.thrift.metastore.WMDropResourcePlanRequest;
+import io.trino.hive.thrift.metastore.WMDropResourcePlanResponse;
+import io.trino.hive.thrift.metastore.WMDropTriggerRequest;
+import io.trino.hive.thrift.metastore.WMDropTriggerResponse;
+import io.trino.hive.thrift.metastore.WMGetActiveResourcePlanRequest;
+import io.trino.hive.thrift.metastore.WMGetActiveResourcePlanResponse;
+import io.trino.hive.thrift.metastore.WMGetAllResourcePlanRequest;
+import io.trino.hive.thrift.metastore.WMGetAllResourcePlanResponse;
+import io.trino.hive.thrift.metastore.WMGetResourcePlanRequest;
+import io.trino.hive.thrift.metastore.WMGetResourcePlanResponse;
+import io.trino.hive.thrift.metastore.WMGetTriggersForResourePlanRequest;
+import io.trino.hive.thrift.metastore.WMGetTriggersForResourePlanResponse;
+import io.trino.hive.thrift.metastore.WMValidateResourcePlanRequest;
+import io.trino.hive.thrift.metastore.WMValidateResourcePlanResponse;
+import io.trino.testing.ResourcePresence;
+import org.apache.thrift.TException;
+import org.apache.thrift.TProcessor;
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.protocol.TProtocolFactory;
+import org.apache.thrift.server.TServlet;
+import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.util.thread.ExecutorThreadPool;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+
+public class MockThriftHttpMetastoreServer
+{
+    private final Server jettyServer;
```

Comment:
Use `TestingHttpServerModule` for this rather than using Jetty directly. See `io.trino.plugin.base.util.TestingHttpServer` for an example.

We'll need a release of libthrift 0.19.0 for this: https://issues.apache.org/jira/browse/THRIFT-5718

Reply:
Unfortunately, I found that the `TestingHttpServerModule` uses the `jakarta.servlet.Servlet` and the `TServlet` which I need to implement in the mock thrift server depends on `javax.servlet.Servlet`. So when I attempted to use the `TestingHttpServerModule` it conflicts with the error which says that the `MockThriftHttpServlet` cannot be cast to `jakarta.servlet.Servlet`

Reply:
Ah, I see what you meant above. the 0.19.0 lib thrift uses jakarta.servlet.Servlet and hence it can work without any additional work-arounds with `TestingHttpServerModule`. Since 0.19.0 may take a while, I worked around this problem by implementing a `TServlet` in the test class itself similar to libthrift's implementation but using the `jakarta.*` packages.